source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -3231,7 +3455,27 @@ export class ChoiceSetInput extends Input {
                 }
             }
 
-            input.onchange = () => { this.valueChanged(); }
+            input.onchange = () => {
+                this.valueChanged();
+                if (choice.inlineAction instanceof ToggleVisibilityAction) {
```

Comment:
I'm not getting this. The `onchange` handler should only do this:
```typescript
input.onchange = () => {
    this.valueChanged();

    if (choice.inlineAction) {
        choice.inlineAction.execute();
    }
}
```

You've already said that this can basically only be a ToggleVisibilityAction in the definition of the inlineAction property, and there is no reason to dynamically change the list of target elements.

`ToggleVisibilityAction` works the following way:
- If a target element is defined as just a string, then the action will hide the element if it is visible and show it if it is hidden
- If a target element's `isVisible` property is explicitly defined, then the action will set the visibility of the element accordingly

In other words, to actually "toggle" the visibility of an element, the associated `targetElement` must not have its `isVisible` property set.

If I apply this principle to the test payload you provided, a card author would have to do the following:

If the ChoiceSet is in multiselect mode:
```
{
  "type": "AdaptiveCard",
  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
  "version": "1.3",
  "body": [
    {
      "id": "ChoiceSet",
      "type": "Input.ChoiceSet",
      "choices": [
        {
          "title": "Choice1",
          "value": "Choice1",
          "inlineAction": {
            "type": "Action.ToggleVisibility",
            "targetElements": [
              {
                "elementId": "FileInput"
              }
            ]
          }
        },
        {
          "title": "Choice2",
          "value": "Choice2",
          "inlineAction": {
            "type": "Action.ToggleVisibility",
            "targetElements": [
              {
                "elementId": "ColorInput"
              }
            ]
          }
        }
      ],
      "value": "",
      "isMultiSelect": true,
      "style": "expanded"
    }
  ]
}
```

If the ChoiceSet is NOT in multiselect mode:
```
{
  "type": "AdaptiveCard",
  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
  "version": "1.3",
  "body": [
    {
      "id": "ChoiceSet",
      "type": "Input.ChoiceSet",
      "choices": [
        {
          "title": "Choice1",
          "value": "Choice1",
          "inlineAction": {
            "type": "Action.ToggleVisibility",
            "targetElements": [
              {
                "elementId": "FileInput",
		"isVisible": true
              },
              {
                "elementId": "ColorInput",
		"isVisible": false
              }
            ]
          }
        },
        {
          "title": "Choice2",
          "value": "Choice2",
          "inlineAction": {
            "type": "Action.ToggleVisibility",
            "targetElements": [
              {
                "elementId": "FileInput",
		"isVisible": false
              },
              {
                "elementId": "ColorInput",
		"isVisible": true
              }
            ]
          }
        }
      ],
      "value": "",
      "isMultiSelect": true,
      "style": "expanded"
    }
  ]
}
```

Reply:
I did some little changes on your suggestion, but thank you for that, i think i got lost trying to find a solution here !